src/lib/ui/noise-sensor/NoiseSensorEventItem.tsx

```diff
@@ -0,0 +1,76 @@
+import type { Event } from 'seamapi'
+
+import { formatDateTime } from 'lib/dates.js'
+import { ClockIcon } from 'lib/icons/Clock.js'
+
+interface NoiseSensorEventItemProps {
+  event: Event
+}
+
+export function NoiseSensorEventItem({
+  event,
+}: NoiseSensorEventItemProps): JSX.Element {
+  const { date, time } = formatDateTime(event.created_at)
+
+  return (
+    <div className='seam-noise-sensor-event-item'>
+      <div className='seam-noise-sensor-event-item-column-wrap'>
+        <ClockIcon />
+
+        <div className='seam-noise-sensor-event-item-datetime-wrap'>
+          <p className='seam-noise-sensor-event-item-date'>{date}</p>
+          <p className='seam-noise-sensor-event-item-time'>{time}</p>
+        </div>
+      </div>
+
+      <div className='seam-noise-sensor-event-item-context-wrap'>
+        <p className='seam-noise-sensor-event-item-context-label'>
+          {getContextLabel(event)}
+        </p>
+        {getContextSublabel(event) != null && (
+          <p className='seam-noise-sensor-event-item-context-sublabel'>
+            {getContextSublabel(event)}
+          </p>
+        )}
+      </div>
+
+      <div className='seam-noise-sensor-event-item-right-block' />
+    </div>
+  )
+}
+
+function getContextLabel(event: Event): string {
+  switch (event.event_type) {
+    case 'noise_sensor.noise_threshold_triggered':
+      return `${t.noiseThresholdTriggered}`
+    case 'noise_detection.detected_noise':
+      return `${t.detectedNoise}`
+    default:
+      return t.noiseDetected
```

Comment:
When would this fallback ever happen?

Reply:
This was for TS satisfaction. I can adjust the approach if needed

Reply:
This should either fallback to null then, or the `Event` type should be narrowed.

I think this means we should provide a better `useEvents` hook, something that would narrow the type based on the `event_types` parameter. This is a good feature for the TS SDK, but we should be able to implement something here for now, even if it's just a generic param, like `useEvents<"noise_sensor.noise_threshold_triggered" | "noise_detection.detected_noise">`

Reply:
I like the generic pattern on `useEvents` 👍. If it's cool with you, I'll move that to a subsequent PR to keep this one relatively isolated (also since the removal of the `detected_noise` event nullifies this specific issue).